x/evm/types/genesis.go

```diff
@@ -1,29 +1,217 @@
 package types
 
 import (
+	"bytes"
 	"encoding/json"
 	"fmt"
+	"sort"
+	"strconv"
+	"strings"
 
+	"github.com/axelarnetwork/axelar-core/utils"
 	"github.com/cosmos/cosmos-sdk/codec"
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 )
 
+// NewGenesisState returns a new genesis state
+func NewGenesisState(chains []GenesisState_Chain) GenesisState {
+	sort.Slice(chains, less(chains))
+
+	return GenesisState{Chains: chains}
+}
+
+func less(chains []GenesisState_Chain) func(i int, j int) bool {
+	return func(i, j int) bool {
+		return chains[i].Params.Chain < chains[j].Params.Chain
+	}
+}
+
 // DefaultGenesisState returns a default genesis state
 func DefaultGenesisState() *GenesisState {
-	return &GenesisState{DefaultParams()}
+	return &GenesisState{Chains: DefaultChains()}
+}
+
+// DefaultChains returns the default chains for a genesis state
+func DefaultChains() []GenesisState_Chain {
+	p := DefaultParams()
+
+	var chains []GenesisState_Chain
+	for _, params := range p {
```

Comment:
```suggestion
	for _, params := range DefaultParams() {
```
what I would do to avoid awkward variable name such as `p`.